src/ui_fsmenu/addons/remote_interaction.cc

```diff
@@ -370,6 +369,108 @@ void CommentEditor::reset_text() {
 	think();
 }
 
+/* TransifexSettingsBox implementation */
+
+class TransifexSettingsBox : public UI::Box {
+public:
+	TransifexSettingsBox(UI::Box& parent, std::shared_ptr<AddOns::AddOnInfo> info)
+	   : UI::Box(&parent, UI::PanelStyle::kFsMenu, 0, 0, UI::Box::Vertical),
+	     priority_(this, 0, 0, 0, 0, UI::PanelStyle::kFsMenu),
+	     name_(this, 0, 0, 450, UI::PanelStyle::kFsMenu),
+	     categories_(this, 0, 0, 0, UI::PanelStyle::kFsMenu) {
+		add(new UI::Textarea(this, UI::PanelStyle::kFsMenu, UI::FontStyle::kFsMenuInfoPanelHeading,
+		                     pgettext("tx", "Priority:"), UI::Align::kCenter),
+		    UI::Box::Resizing::kFullSize);
+		add_space(kRowButtonSpacing);
+		add(&priority_, UI::Box::Resizing::kFullSize);
+		add_space(kRowButtonSpacing);
+		add(new UI::Textarea(
+		       this, UI::PanelStyle::kFsMenu, UI::FontStyle::kFsMenuInfoPanelHeading,
+		       /** TRANSLATORS: "Resource" here refers to the name of a translation unit */
+		       pgettext("tx", "Resource Name:"), UI::Align::kCenter),
+		    UI::Box::Resizing::kFullSize);
+		add_space(kRowButtonSpacing);
+		add(&name_, UI::Box::Resizing::kFullSize);
+		add_space(kRowButtonSpacing);
+		add(new UI::Textarea(this, UI::PanelStyle::kFsMenu, UI::FontStyle::kFsMenuInfoPanelHeading,
+		                     pgettext("tx", "Categories (whitespace-separated; characters only):"),
+		                     UI::Align::kCenter),
+		    UI::Box::Resizing::kFullSize);
+		add_space(kRowButtonSpacing);
+		add(&categories_, UI::Box::Resizing::kFullSize);
+		add_space(kRowButtonSpacing);
+		add(new UI::Textarea(this, UI::PanelStyle::kFsMenu, UI::FontStyle::kFsMenuInfoPanelHeading,
+		                     _("This may take several minutes. Please be patient."),
+		                     UI::Align::kCenter),
+		    UI::Box::Resizing::kFullSize);
+
+		name_.set_text(info->unlocalized_descname);
+		priority_.add(_("Low"), "normal", nullptr, false);
+		priority_.add(_("Normal"), "high", nullptr, true);
+		priority_.add(_("High"), "urgent", nullptr, false);
+		priority_.set_desired_size(300, priority_.get_lineheight() * (priority_.size() + 1));
+	}
+
+	/** Whether the data is valid and can be submitted. */
+	bool ok_enabled() const {
+		return priority_.has_selection() && !name_.text().empty() &&
+		       categories_.text().find_first_not_of(
+		          " abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ") == std::string::npos;
+	}
+
+	/**
+	 * Generate the data for the server command, in the format
+	 * "Priority <Linefeed> Name <Linefeed> [Categories]".
+	 */
+	std::string make_data() const {
+		std::string str = priority_.get_selected();
+		str += '\n';
+		str += name_.text();
+		str += "\n[";
+
+		std::string cats = categories_.text();
+		size_t count = 0;
+		for (;;) {
```

Comment:
Suggestion for replacing the loop using regex magic
```
  std::regex rex_cat("( *)([a-zA-Z]+)( +|$)");
  str += std::regex_replace(cats, rex_cat, "\"$2\",");
  str.pop_back(); // strip last ,
```
Try it:
<details>
<pre>
#include <iostream>
#include <regex>
#include <string>

int main(){
  std::string str = " tribes  world";
  std::regex rex_cat("( *)([a-zA-Z]+)( +|$)");
  
  std::cout << str << std::endl;
  std::cout << std::regex_replace(str, rex_cat, "\\"$2\\",") << std::endl;
}
</pre>
</details>